build_mt.py

```diff
@@ -177,38 +177,29 @@ def correct_inds(halo_ids, N_halos_slabs, slabs, inds_fn):
 
     return ids
 
-def get_mt_info(fns, fields, minified, inds_fn):
+def get_mt_info(fns_load, fields, minified):
     '''
     Load merger tree and progenitors information
     '''
-
-    # selected files to load
-    fns_load = [fns[i] for i in inds_fn]
-
-    # if we are loading all progenitors and not just main
-    if "Progenitors" in fields:
-        merger_tree, Progs = simple_load(fns_load, fields=fields)
-    else:
-        merger_tree = simple_load(fns_load, fields=fields)
-
-    # turn data into astropy table
-    Merger = Table(merger_tree, copy=False)
-    Merger.add_column('ComovingDistance', copy=False)
+    
+    data = simple_load(fns_load, fields=fields)
+    merger = data['merger']
```

Comment:
I tried to remove all the "Progenitors" special-casing; I was having some trouble following the algorithm, and I'm hoping reducing the code verbosity and repetition will help with that.  Instead, the `simple_load()` function returns a dict that may or may not have a `progenitors` entry.  That way, you don't need two separate function calls, one that takes one return value, and one that takes two.